sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1610,6 +1610,13 @@ object SQLConf {
       """ "... N more fields" placeholder.""")
     .intConf
     .createWithDefault(25)
+
+  val LEGACY_DECIMAL_PARSING_ENABLED = buildConf("spark.sql.legacy.decimalParsing.enabled")
+    .doc("If it is set to false, it enables parsing decimals in locale specific formats. " +
+      "To switch back to previous behaviour when parsing was performed by java.math.BigDecimal " +
+      "and all commas were removed from the input, set the flag to true.")
+    .booleanConf
+    .createWithDefault(false)
```

Comment:
Can we have `true` by default? I usually recommend to keep the existing behavior during one next release for safety. But, @gatorsmile might have different opinion for this.

Reply:
> Can we have true by default?

Yes, sure. 

>  I usually recommend to keep the existing behavior during one next release for safety.

Would it be better to switch on new implementation in major release (3.0) neither in minor one?